audio/descriptors.c

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (C) 2017 Kyle Robbertze
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
```

Comment:
It is based on the bulkloop example, which is GPL-v3 - https://github.com/mithro/fx2lib/blob/linux-descriptors/COPYING